### PR TITLE
Fix compilation error due to exception-type-mismatch

### LINKED ISSUE #289: Generated code is broken due to mismatch with hand-written `server.h`
At [`eventuals/grpc/server.h:54`](https://github.com/3rdparty/eventuals/blob/35e1fcf4e197f0c3fa72179b19db4c6f0fd672cb/eventuals/grpc/server.h#L54) we find the following:
```cpp
class Service {
 public:
  // [...]
  virtual Task::Of<void>::Raises<std::runtime_error> Serve() = 0;
```

While since #282 the code generated by `protoc-gen-eventuals` looks as follows:
```cpp
class TypeErasedService : public ::eventuals::grpc::Service {
 public:
  ::eventuals::Task::Of<void>::Raises<std::exception> Serve() override;
```

The mismatch between `std::runtime_error` and `std::exception` causes compilation failures, such as in https://github.com/reboot-dev/respect/issues/230#issuecomment-1082196695:
```
In file included from ./object-store/object-store.h:8,
                 from object-store/main.cc:13:
bazel-out/k8-fastbuild/bin/external/com_github_reboot_dev_respect_api/respect/v1/object_store.eventuals.h:20:57: error: invalid covariant return type for 'virtual eventuals::_Task<void, void, std::tuple<> >::Raises<std::exception> respect::v1::eventuals::ObjectStore::TypeErasedService::Serve()'
   20 |     ::eventuals::Task::Of<void>::Raises<std::exception> Serve() override;
      |                                                         ^~~~~
In file included from object-store/main.cc:8:
external/com_github_3rdparty_eventuals/eventuals/grpc/server.h:54:54: note: overridden function is 'virtual eventuals::_Task<void, void, std::tuple<> >::Raises<std::runtime_error> eventuals::grpc::Service::Serve()'
   54 |   virtual Task::Of<void>::Raises<std::runtime_error> Serve() = 0;
      |                                                      ^~~~~
Target //object-store:image.tar failed to build
```

### PR DESCRIPTION
**Pull request by rjhuijsman**
As reported in https://github.com/3rdparty/eventuals/issues/289 our
generated code fails to compile due to a mismatch between hand-written
code in `server.h` (which uses `std::runtime_error`) and auto-generated
code (which uses `std::exception`).

This PR resolves that conflict in favor of the hand-written code,
changing the generated code to use `std::runtime_error`. It picks this
approach since... I couldn't get the other way 'round to compile! I'd
love to chat with you, @benh, to learn why that is.

In the meantime, this demonstrates at least one possible fix. I've
confirmed that our Respect object store compiles with these latest
fixes, resolving https://github.com/reboot-dev/respect/issues/230.

TESTED: compiled the Respect object store.

Fixes #289

### CODE REVIEW

**Review by CodingCanuck — APPROVED** [review]
Drive by LGTM, but feel free to wait for @benh 's approval

**Comment by benh** [comment]
@rjhuijsman can you share the error you got when trying to return `std::exception`?

**Comment by CodingCanuck** [comment]
> @rjhuijsman can you share the error you got when trying to return `std::exception`?

I think the error is here: https://github.com/3rdparty/eventuals/issues/289

Specifically:
```
In file included from ./object-store/object-store.h:8,
                 from object-store/main.cc:13:
bazel-out/k8-fastbuild/bin/external/com_github_reboot_dev_respect_api/respect/v1/object_store.eventuals.h:20:57: error: invalid covariant return type for 'virtual eventuals::_Task<void, void, std::tuple<> >::Raises<std::exception> respect::v1::eventuals::ObjectStore::TypeErasedService::Serve()'
   20 |     ::eventuals::Task::Of<void>::Raises<std::exception> Serve() override;
      |                                                         ^~~~~
In file included from object-store/main.cc:8:
external/com_github_3rdparty_eventuals/eventuals/grpc/server.h:54:54: note: overridden function is 'virtual eventuals::_Task<void, void, std::tuple<> >::Raises<std::runtime_error> eventuals::grpc::Service::Serve()'
   54 |   virtual Task::Of<void>::Raises<std::runtime_error> Serve() = 0;
      |                                                      ^~~~~
Target //object-store:image.tar failed to build
```

**Comment by rjhuijsman** [comment]
@benh found the bug that was causing me strange compilation errors, so the new approach here is https://github.com/3rdparty/eventuals/pull/292

**Pull request closed without merging**
